### PR TITLE
[PR] Remove unused debug/dump code.

### PR DESCRIPTION
**Pull request by jeremyfelt**
Anyone debugging their JavaScript application should be able to implement something specific to their needs. We aren't using this anywhere in the Spine right now.

### CODE REVIEW

**Comment by jeremyBass** [comment]
I would have to take a look at why the flag is not working now.. all tests and build were divided in to a debug and production, using the node package `env` iirc.  You then set in the task the value, so if the build is skipping the value it may default to doing nothing.  Possible reason for it failing.  

What it solves is that when you do local build and do all the cross browser and mobile checks, not always is the implementation of `console` consistent (in the past even missing thought that is not the case now I'm pretty sure).  What that function does is basically what you can see in chrome, where you dump arrays/object for example you can toggle the data tree in the console so see the live values.  It is even helpful in chrome as you have limits in the console that is needed to get around.  Also some times in the implementation of the ECMAScript engines for a product like android or something, there is flaws where the data doesn't come out correctly and you get in the area where you can't fully log the object as you need to, so you can use this to call and get it more robustly.

Another part was that in the unit tests that should be done at some point, this was also going to be used to print as well.  That is a side note.

In short what this has solved is those hard to see issues before on devices that you can't always get the output for directly.  I personally haven't needed to use in a while, but it was left for unit tests and so people didn't go reinventing the wheel to solve that edge case, and since it's in the debug it was better to keep the tool handy then to be caught needing it.  It shouldn't be coming out in production.
